client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/list-item.js

```diff
@@ -0,0 +1,166 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+import { trim } from 'lodash';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import CarrierIcon from '../../components/carrier-icon';
+import Dialog from 'components/dialog';
+import {
+	getCarrierAccountsState,
+} from 'woocommerce/woocommerce-services/state/carrier-accounts/selectors';
+import {
+	setVisibilityDisconnectCarrierDialog,
+	disconnectCarrier,
+} from 'woocommerce/woocommerce-services/state/carrier-accounts/actions';
+
+const CarrierAccountListItem = ( props ) => {
+	const {
+		isPlaceholder,
+		data,
+		children,
+		translate,
+		siteId,
+		showDisconnectDialog,
+	} = props;
+	if ( isPlaceholder ) {
+		return (
+			<div className="carrier-accounts__list-item">
+				<div className="carrier-accounts__list-item-placeholder-carrier-icon">
+					<div className="carrier-accounts__list-item-carrier-icion-placeholder">
+						<span />
+					</div>
+				</div>
+				<div className="carrier-accounts__list-item-name">
+					<div className="carrier-accounts__list-item-name-placeholder">
+						<span />
+					</div>
+				</div>
+				<div className="carrier-accounts__list-item-actions">{ children }</div>
+			</div>
+		);
+	}
+
+	const renderIcon = carrierId => {
+		return <div className="carrier-accounts__list-item-carrier-icon">
+			<CarrierIcon carrier={ carrierId } size={ 18 } />
+		</div>;
+	};
+
+	const renderName = name => {
+		const carrierName = name && '' !== trim( name ) ? name : translate( 'Untitled' );
+		return <div className="carrier-accounts__list-item-name">
+			<span>{ carrierName }</span>
+		</div>;
+	};
+
+	const renderCredentials = credentials => {
+		return <div className="carrier-accounts__list-item-credentials">
+			<span>{ credentials }</span>
+		</div>;
+	}
+
+	const showDisconnectDialogHandler = () => {
+		props.setVisibilityDisconnectCarrierDialog( siteId, data.carrierId, true );
+	}
+
+	const hideDisconnectDialogHandler = () => {
+		props.setVisibilityDisconnectCarrierDialog( siteId, data.carrierId, false );
+	}
+
+	const renderActions = () => {
+		const { credentials,  onConnect, onDisconnect } = data;
```

Comment:
My lint console complaint: `84:36  error  'onDisconnect' is assigned a value but never used  no-unused-vars` :D